File: 41firstMissingPositive/firstMissingPositive.py
class Solution:
    def firstMissingPositive(self, nums: List[int]) -> int:
        for index in range(len(nums)):
            while 0 <= nums[index] - 1 < len(nums) and nums[nums[index] - 1] != nums[index]: nums[nums[index] - 1], nums[index] = nums[index], nums[nums[index] - 1]
        for index in range(len(nums)):
            if nums[index] != index + 1: return index + 1
        return len(nums) + 1

File: 41firstMissingPositive/firstMissingPositive.java
class Solution {
    public int firstMissingPositive(int[] nums) {
        java.util.stream.IntStream.range(0, nums.length).forEach(index -> {
            while (0 <= nums[index] - 1 && nums[index] - 1 < nums.length && nums[nums[index] - 1] != nums[index])
            {
                int temp = nums[nums[index] - 1];
                nums[nums[index] - 1] = nums[index];
                nums[index] = temp;
            }});
        for (int index = 0; index != nums.length ; ++index)
            if (nums[index] != index + 1) return index + 1;
        return nums.length + 1;        
    }
}
